thrust/system/cuda/detail/transform_scan.h

```diff
@@ -50,8 +50,8 @@ transform_inclusive_scan(execution_policy<Derived> &policy,
                          TransformOp                transform_op,
                          ScanOp                     scan_op)
 {
-  // Use the input iterator's value type per https://wg21.link/P0571
-  using result_type = typename thrust::iterator_value<InputIt>::type;
+  using input_type = typename thrust::iterator_value<InputIt>::type;
+  using result_type = typename std::result_of<TransformOp(input_type)>::type;
```

Comment:
`std::result_of` is deprecated in C++17. Can you update this to use the same pattern as `transform_input_output_iterator`:

https://github.com/thrust/thrust/blob/767647039fc78eccaa643a5d89b3bc863bb48a0f/thrust/iterator/detail/transform_input_output_iterator.inl#L35-L40

Reply:
Done.